doc/API.md

```diff
@@ -507,6 +517,311 @@ const key = '/key'
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Array<multiaddrInfo>` | Array of peer's multiaddr with their relevant information |
+
+#### Example
+
+```js
+peerStore.addressBook.set(peerId, multiaddr)
+// [
+// {
+//   multiaddr: /ip4/140.10.2.1/tcp/8000,
+//   ...
+// },
+// {
+//   multiaddr: /ip4/140.10.2.1/ws/8001
+//   ...
+// },
+// ]
+```
+
+### peerStore.protoBook.delete
+
+Delete the provided peer from the book.
+
+`peerStore.protoBook.delete(peerId)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | [`PeerId`][peer-id] | peerId to remove |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `boolean` | true if found and removed |
+
+#### Example
+
+```js
+peerStore.protoBook.delete(peerId)
+// false
+peerStore.protoBook.set(peerId, protocols)
+peerStore.protoBook.delete(peerId)
+// true
+```
+
+### peerStore.protoBook.get
+
+Get the known `protocols` of a provided peer.
+
+`peerStore.protoBook.get(peerId)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | [`PeerId`][peer-id] | peerId to get |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Array<string>` | Array of peer's supported protocols |
+
+#### Example
+
+```js
+peerStore.protoBook.get(peerId)
+// undefined
+peerStore.protoBook.set(peerId, [ '/proto/1.0.0', '/proto/1.1.0' ])
+peerStore.protoBook.get(peerId)
+// [ '/proto/1.0.0', '/proto/1.1.0' ]
+```
+
+### peerStore.protoBook.set
+
+Set known `protocols` of a given peer.
+
+`peerStore.protoBook.set(peerId, protocols, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | [`PeerId`][peer-id] | peerId to set |
+| protocols | `string|Array<string>` | protocols to store |
+| [options] | `Object` | options to set |
+| [options.replace] | `Object` | replace stored data (if exists) or unique union (default: true) |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Array<string>` | Array of peer's supported protocols |
+
+#### Example
+
+```js
+peerStore.protoBook.set(peerId, supportedProtocols)
+// [ '/proto/1.0.0', '/proto/1.1.0' ]
+```
+
+### peerStore.delete
+
+Delete the provided peer from every book.
+
+`peerStore.delete(peerId)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peerId | [`PeerId`][peer-id] | peerId to remove |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `boolean` | true if found and removed |
+
+#### Example
+
+```js
+peerStore.delete(peerId)
+// false
+peerStore.addressBook.set(peerId, discoveredMultiaddrs)
+peerStore.protoBook.set(peerId, supportedProtocols)
+peerStore.delete(peerId)
+// true
+peerStore.delete(peerId2)
+// false
+peerStore.addressBook.set(peerId2, discoveredMultiaddrs)
+peerStore.delete(peerId2)
+// true
+```
+
+### peerStore.find
+
+Find the stored information of a given peer.
+
+`peerStore.find(peerId)`
```

Comment:
Should we just call this `peerStore.get(peerId)`? I'm worried users might misinterpret this as a network search for the peer.

Reply:
Yes, I agree! The question is, should I do it now, or in the next PR? (this will break the return value)